oda_api/api.py

```diff
@@ -936,13 +936,23 @@ def _decode_res_json(self, res):
 
             raise RemoteException(message=msg)
 
+    def get_token_from_environment(self):
+        token = oda_api.token.discover_token(self.token_discovery_methods)
+        if token is not None:
+            logger.info("discovered token in environment")
+
+        return token
+
     @safe_run
-    def get_instrument_description(self, instrument=None):
+    def get_instrument_description(self, instrument=None, token=None):
         if instrument is None:
             instrument = self.instrument
 
+        if token is None:
```

Comment:
this is repeated. Is it not possible to put it in the constructor?

Reply:
do we want to have it in the constructor? I am not sure is gonna be a smooth change, I will try and see

Reply:
token is now set in the constructor